src/GraphQL/Execution/InputsExtensions.cs

```diff
@@ -0,0 +1,15 @@
+using System.Collections.Generic;
+
+namespace GraphQL
+{
+    public static class InputsExtensions
+    {
+        /// <summary>
+        /// Converts a dictionary into an <see cref="Inputs"/>.
+        /// </summary>
+        /// <param name="json">A dictionary.</param>
+        /// <returns>Inputs.</returns>
+        public static Inputs ToInputs(this Dictionary<string, object> dictionary)
```

Comment:
I think that this method has a little value. Nevertheless, we can change it a little. I propose to make `Inputs` immutable:

```c#
 public class Inputs : ReadOnlyDictionary<string, object>
    {
        public static readonly Inputs Empty = new Inputs(new Dictionary<string, object>());

        public Inputs(IDictionary<string, object> dictionary)
            : base(dictionary)
        {
        }
    }
``` 

No default ctor + public Empty. Then in `InputExtensions`:
```c#
 public static Inputs ToInputs(this Dictionary<string, object> dictionary)
            => dictionary == null ? Inputs.Empty : new Inputs(dictionary);
```

Then only one edit in the `StarWarsBasicQueryTests` is required:
```c#
  var inputs = new Inputs(new Dictionary<string, object> {{"id", "1"}});
```

This will allow not to allocate a new `Dictionary` for variables to each request when there are no variables.

Reply:
Great idea. Done.